### ISSUE TITLE
Records with certain data may irrecoverably break the database

### ISSUE THREAD

**Issue by MadCatX** [Molecular-Biophysics-Database/mbdb-app#4]
An API call to create a new MST record with the following payload:

```
{
	"metadata": {
		"general_parameters": {
			"associated_publications": {
				"additional": [],
				"main": {
					"authors": [],
					"resource_type": "Article"
				}
			},
			"chemical_information": {
				"chemical_environments": [],
				"entities_of_interest": []
			},
			"depositors": {
				"contributors": [],
				"depositor": {
					"affiliations": [],
					"identifiers": []
				},
				"principal_contact": {
					"affiliations": [],
					"identifiers": []
				}
			},
			"derived_parameters": [],
			"funding_reference": [],
			"instrument": {
				"manufacturer": "Bio-Rad",
				"performance_test": {
					"published_test_protocol": {
						"authors": [],
						"resource_type": "Article"
					},
					"report": {
						"processing_steps": [],
						"size": {
							"unit": "bytes (B)"
						},
						"source": "Instrument software",
						"type": "text"
					},
					"sample_composition": []
				}
			},
			"physical_environment_at_sample_handling": {
				"atmosphere": {
					"composition": []
				},
				"humidity": {
					"controlled": false,
					"obtained_by": "Measurement",
					"unit": "%",
					"value_error": {
						"errors_are_relative": false
					}
				},
				"pressure": {
					"controlled": false,
					"obtained_by": "Measurement",
					"unit": "Pa",
					"value_error": {
						"errors_are_relative": false
					}
				},
				"temperature": {
					"controlled": false,
					"obtained_by": "Measurement",
					"operational_value": "Room temperature",
					"unit": "K",
					"value_error": {
						"errors_are_relative": false
					}
				}
			},
			"raw_data_information": {
				"collection_start_time": "2023-06-25",
				"file_information": []
			},
			"record": {
				"access_rights": "open access",
				"date_available": "2023-06-25",
				"deposition_date": "2023-06-25",
				"keywords": [],
				"metadata_access_rights": "open access",
				"project": {
					"owner": {
						"affiliations": [],
						"identifiers": []
					}
				},
				"publisher": "MBDB",
				"resource_type_general": "dataset"
			},
			"technique": "Bio-layer interferometry (BLI)"
		},
		"method_specific_parameters": {
			"data_analysis": [],
			"excitation_led_color": "RED (ex 605–645nm, em 660–730nm)",
			"experiment_type": "Affinity",
			"label_free": false,
			"measurements": [
				{
					"measured_data": {
						"x_data": {
							"id": "46cb7471-7128-4459-936f-c5c8bc35e79f",
							"values": []
						},
						"y_data": {
							"id": "1429bea7-9bd6-4758-8217-698b5ea48e5a",
							"values": []
						}
					},
					"sample": {
						"container": "Monolith Standard Capillary",
						"ligands": [],
						"preparation": [],
						"targets": []
					}
				}
			],
			"signal_type": "Initial intensity"
		}
	}
}
```

ends up with the server returning 500. A long backtrace pointing to the `marshmallow` package ends with
```
TypeError: 'builtin_function_or_method' object is not iterable 
```

The offending part seems to be the `measurements`. If it's left empty, the otherwise same payload does not trigger any problems.

This would not be so much of a problem if the problematic data didn't get stored in the DB. They, however, apparently do because all subsequent attempts to display listing of all MST data fails with a 500 and the same backtrace.

**Comment by edager** [comment]
According to @mesemus this was caused by: "the problem is inside the marshmallow schema - python implementation gets the value from a dictionary using [] at first and if it fails it uses getattr. If a name of a property is a name of a method (for example, "values" inside xdata) and it is not filled in inside the data, marshmallow will try to serialize the method instead and will fail. I'll add a workaround to the model builder to fix this. "

This should have been fixed with #83  so I'm closing it

**Comment by edager** [comment]
According to @mesemus this was caused by: "the problem is inside the marshmallow schema - python implementation gets the value from a dictionary using [] at first and if it fails it uses getattr. If a name of a property is a name of a method (for example, "values" inside xdata) and it is not filled in inside the data, marshmallow will try to serialize the method instead and will fail. I'll add a workaround to the model builder to fix this. "

This should have been fixed with #83 so I'm closing it.
